src/simulator/views/map/display/solid_colour_map_display_pot.py

```diff
@@ -24,35 +25,36 @@ def __init__(self, services: Services, points, z_index=50, radius=0, custom_map:
         self.large = max(pmaplst)
         self.small = min(pmaplst)
         self.range = (self.large-self.small)/6
+        self.pts = set()
 
         #assert self._map.size.n_dim == 2, "Unsupported map dimension, expected 2D"
 
-    def render(self) -> bool:
-        if not super().render():
-            return False
-
+    def render(self, refresh: bool) -> None:
         points: Union[Set[Point], List[Entity]] = copy.deepcopy(self.points)
         if points is None:
-            return False
+            return
 
         def f(pt):
             if isinstance(pt, Entity):
                 pt = pt.position
             return pt
 
-        points = set(map(f, points))
-
-        for point in points:
-            if point[1] < self.small+(self.range*1):
-                self._root_view.render_pos(Entity(point[0], self.radius), self.color6)
-            elif point[1] < self.small+(self.range*2):
-                self._root_view.render_pos(Entity(point[0], self.radius), self.color5)
-            elif point[1] < self.small+(self.range*3):
-                self._root_view.render_pos(Entity(point[0], self.radius), self.color4)
-            elif point[1] < self.small+(self.range*4):
-                self._root_view.render_pos(Entity(point[0], self.radius), self.color3)
-            elif point[1] < self.small+(self.range*5):
-                self._root_view.render_pos(Entity(point[0], self.radius), self.color2)
-            elif point[1] < self.small+(self.range*6):
-                self._root_view.render_pos(Entity(point[0], self.radius), self.color1)
-        return True
+        self.pts = set(map(f, points))
+
+        for p in self.pts:
+            self._root_view.cube_requires_update(Entity(p[0], self.radius))
+
+    def update_cube(self, p: Point) -> None:
```

Comment:
is this meant to be a sort of gradient coloring functionality

Reply:
Don't really know and don't really care xD - never tried to understand the logic